kornia/losses/focal.py

```diff
@@ -198,18 +204,25 @@ def binary_focal_loss_with_logits(
             stacklevel=2,
         )
 
-    if not isinstance(input, torch.Tensor):
-        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+    if not isinstance(input, Tensor):
+        raise TypeError(f"Input type is not a Tensor. Got {type(input)}")
 
     if not len(input.shape) >= 2:
         raise ValueError(f"Invalid input shape, we expect BxCx*. Got: {input.shape}")
 
     if input.size(0) != target.size(0):
         raise ValueError(f'Expected input batch_size ({input.size(0)}) to match target batch_size ({target.size(0)}).')
 
+    if pos_weight is None:
+        pos_weight = torch.ones(input.size(-1), device=input.device, dtype=input.dtype)
```

Comment:
I suggest a modification of `input.size(-1)` because of a pytorch-style input size with

![image](https://user-images.githubusercontent.com/78028879/177000411-1c623436-4ac3-46fc-b47d-356f7e6265b6.png)

in [torch.nn.CrossEntropyLoss](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html#torch.nn.CrossEntropyLoss), the same as other places like `pos_weight` size check.

Can also add a doc to describe the input size.

Reply:
@xiaoyuan0203 You are right. I added you as collaborator in my forked branch (I couldn't push my changes without it into your forked repo), so you can make your changes to `input.size(-1)`

Reply:
I think that it is advisable to use here only general constructions from PyTorch and use more numpy/jax style. 

What you say, may be changed here:
[https://github.com/facebookresearch/fvcore/blob/main/fvcore/nn/focal_loss.py#L36](https://github.com/facebookresearch/fvcore/blob/main/fvcore/nn/focal_loss.py#L36)

The `F.binary_cross_entropy_with_logits(inputs, targets, reduction="none")` is already used [here](https://github.com/facebookresearch/fvcore/blob/main/fvcore/nn/focal_loss.py#L36), so it can be easy add pos_weight as argument.